### DISCUSSION TITLE
commits are showing up under wrong GitHub account

### DISCUSSION THREAD

**Discussion by wardrip** [community/community#69218] [New to GitHub] [answered]
I have cloned my work's private repo using my ssh key.  I've done this many times before without issue, except the latest time I did this.   I created a new branch from master, made a couple small changes, committed and pushed.  On Github, I used the UI to open a PR and the PR shows my correct account, but each commit is under a different GitHub user.  The user the commits are under is https://github.com/RobertusvT.  I then attempted to research any issues that others may have had and I came across Github's documents for verifying commits here.
[About Commit Signature Verification](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification)
The doc notes that I should check for verified, Unverified and No verification status.  but there are no labels at on on each of the commits.
I went to my settings to verify my ssh key and I had no ssh keys set up.  I know I didn't remove them and It was set up because that is what I used to clone the repo.  I'm wondering if anyone can help me figure out what has happened.

**Comment by AnkurBansal18** [comment, marked as answer]
Hey @wardrip, you can try these steps : 

**Check Your Git Configurations:**

Ensure that your local Git configurations are set correctly. You can check your Git username and email by running the following commands in your terminal:

**git config --get user.name
git config --get user.email**

Make sure that the output of these commands displays your correct GitHub username and the email associated with your GitHub account. If they don't match, you can set them using:

git config --global user.name "Your GitHub Username"
git config --global user.email "Your GitHub Email"

**Verify SSH Key:**

Confirm that your SSH key is properly configured and associated with your GitHub account. To do this, follow these steps:

a. Run the following command to list your SSH keys:

**ls -al ~/.ssh**

b. Check if you have an SSH key named id_rsa or id_rsa.pub (these are the default key names). If not, you may need to generate a new SSH key and add it to your SSH agent. You can generate a new SSH key using ssh-keygen.

c. If you have an SSH key, add it to your SSH agent using the following command:

**ssh-add ~/.ssh/id_rsa**

d. Copy the contents of your public key (id_rsa.pub) and add it to your GitHub account. You can do this by going to your GitHub account settings > SSH and GPG keys and adding a new SSH key.

Check Commit Authorship:

If you're still facing the issue of commits being associated with a different GitHub user, double-check your local Git configuration for any specific authorship settings. You can view your Git configuration by running:

**git config --list**

Look for any entries related to authorship, and if you find any that might be causing the issue, you can remove or modify them accordingly.

Reauthenticate on GitHub:

It's possible that your GitHub authentication token or session may have expired. Try re-authenticating by running:

git pull

This should prompt you to enter your GitHub username and password or use a personal access token if you have one. Follow the prompts to re-authenticate.

Review Your SSH Key Configuration:

Recheck the SSH key you're using by running:

git remote -v
Ensure that the remote URL is in the SSH format (e.g., git@github.com:username/repo.git) and not using HTTPS.

additionally, Review GitHub Actions or Hooks:

If you're using GitHub Actions or any pre-commit hooks, review them to ensure they're not altering the authorship of your commits.
If after following these steps, you're still experiencing issues, consider contacting your company's IT or DevOps team, as there might be specific configurations or policies within your organization that could affect how commits are attributed.
